activerecord/lib/active_record/validations/uniqueness.rb

```diff
@@ -86,12 +86,21 @@ def build_relation(klass, attribute, value) # :nodoc:
       def scope_relation(record, relation)
         Array(options[:scope]).each do |scope_item|
           if reflection = record.class._reflect_on_association(scope_item)
```

Comment:
Can we instead just replace this whole `if` with something like the below?

```ruby
scope_value = record.send(scope_item)
```

`where` knows how to handle an association just fine.. it doesn't seem like something we should be handling here.